src/biogeophys/OzoneLunaMod.F90

```diff
@@ -0,0 +1,451 @@
+       forc_ozone = forc_po3(c) / forc_pbot(c)
+
+       ! Ozone uptake for shaded leaves
+       call this%CalcOzoneUptakeOnePoint( &
+            forc_ozone=forc_ozone, forc_pbot=forc_pbot(c), forc_th=forc_th(c), &
+            rs=rssha(p), rb=rb(p), ram=ram(p), &
+            tlai=tlai(p), tlai_old=tlai_old(p), pft_type=patch%itype(p), &
+            o3uptake=o3uptakesha(p))
+
+       ! Ozone uptake for sunlit leaves
+       call this%CalcOzoneUptakeOnePoint( &
+            forc_ozone=forc_ozone, forc_pbot=forc_pbot(c), forc_th=forc_th(c), &
+            rs=rssun(p), rb=rb(p), ram=ram(p), &
+            tlai=tlai(p), tlai_old=tlai_old(p), pft_type=patch%itype(p), &
+            o3uptake=o3uptakesun(p))
+
+       tlai_old(p) = tlai(p)
+
+    end do
+
+    end associate
+
+    ! Explicitly set outputs to 1. This isn't really needed, because they should still be
+    ! at 1 from cold-start initialization, but do this for clarity here.
+
+    this%o3coefvsha_patch(bounds%begp:bounds%endp) = 1._r8
+    this%o3coefvsun_patch(bounds%begp:bounds%endp) = 1._r8
+    this%o3coefgsha_patch(bounds%begp:bounds%endp) = 1._r8
+    this%o3coefgsun_patch(bounds%begp:bounds%endp) = 1._r8
+
+  end subroutine CalcOzoneStress
+
+!-----------------------------------------------------------------------  
+  subroutine Acc24_OzoneStressOnePoint_Luna(pft_type, o3uptake, o3coefjmax)
+
+    ! !USES:
+    use pftconMod            , only : pftcon
+    !
+    ! ! ARGUMENTS:
+    integer  , intent(in)    :: pft_type    ! vegetation type, for indexing into pftvarcon arrays
+    real(r8) , intent(in)    :: o3uptake    ! ozone entering the leaf
+    real(r8) , intent(inout) :: o3coefjmax  ! ozone coefficient for max. electron transport rate
+    !
+    ! !LOCAL VARIABLES:
+    real(r8) :: jmaxInt        ! intercept for max. electron transport rate
+    real(r8) :: jmaxSlope      ! slope for max. electron transport rate
+
+    if ( o3uptake == 0._r8 ) then
+       ! No o3 damage if no o3 uptake
+       o3coefjmax = 1._r8
+    else
+       ! Determine parameter values for this pft
+       ! TODO(wjs, 2014-10-01) Once these parameters are moved into the params file, this
+       ! logic can be removed.
+       if (pft_type>3) then
+          if (pftcon%woody(pft_type)==0) then
+             jmaxInt    = nonwoodyJmaxInt
+             jmaxSlope  = nonwoodyJmaxSlope
+          else
+             jmaxInt    = broadleafJmaxInt
+             jmaxSlope  = broadleafJmaxSlope
+          end if
+       else
+          jmaxInt    = needleleafJmaxInt
+          jmaxSlope  = needleleafJmaxSlope
+       end if
+       ! Apply parameter values to compute o3 coefficients
+       o3coefjmax = max(0._r8, min(1._r8, jmaxInt  + jmaxSlope  * o3uptake))
+       
+    end if
+
+
+  end subroutine Acc24_OzoneStressOnePoint_Luna
+
+!-----------------------------------------------------------------------  
+  subroutine Acc24_OzoneStress_Luna(this, bounds, num_exposedvegp, filter_exposedvegp)
```

Comment:
The 24 hour accumulation happens within CalcOzoneUptake. Hence the Acc24_OzoneStressOnePoint_Luna is only meant to reflect that it is an **accumulated stress** over the whole day **rather than an instantaneous** one. Or in other words that LUNA is only updated with ozone stress every 24 hours.
This is probably in conflict with your proposed change to clm_driver.f90.

Reply:
Let me see if I can explain my thinking, then you can tell me if you see a problem with this, for example, as it relates to my proposed change to the driver:

If I remember correctly (@danicalombardozzi can correct me if I'm wrong): Even in Danica's original implementation, ozone stress is based on the cumulative uptake since the leaves first came into existence (with some decay & healing offsetting the cumulative uptake). Your version and Danica's seem to calculate ozone uptake the same way. Your new ozone stress still looks based on this accumulated ozone uptake since the start of the leaves' lifetime - e.g., there is no resetting of ozone uptake after each 24-hour period. (To me, a name like Acc24_XXX implies some resetting every 24 hours.) True, the LUNA version of ozone stress only needs to be called at the end of each day because that's when the relevant LUNA routine is called, but I don't see that that changes the nature of the stress calculation fundamentally.

To say this a different way: My understanding is that the model would give the same results if we called your new stress routine every time step rather than just once a day; it's just that we'd be wasting a lot of time doing calculations we don't really need to do. I think that, if the stress at the end of day 1 is X and the stress at the end of day 2 is Y, then typically, the stress calculated in between those two times would vary gradually from X to Y. It's just that LUNA doesn't happen to need the stress value at all of those intermediate times.

What I suggested in my companion PR was that we still only call the new stress routine at the end of the day, though again, I believe this is just a performance booster, not something fundamentally required by the code.

If you feel I'm missing something here, it might be most efficient for us to have a call to discuss this - ideally at a time when Danica can also attend.

Reply:
@billsacks You are absolutely right. It's only that our interpretation of the name Acc24 differs. That's what I meant to say. I don't see any problem renaming the routine accordingly.